pkg/query/iter.go

```diff
@@ -325,10 +325,10 @@ func (s *dedupSeriesSet) Next() bool {
 // replica label if it exists
 func (s *dedupSeriesSet) peekLset() labels.Labels {
 	lset := s.peek.Labels()
-	if lset[len(lset)-1].Name != s.replicaLabel {
+	if _, ok := s.replicaLabels[lset[len(lset)-1].Name]; !ok {
```

Comment:
I would have expected that we need to verify each replica label here. I feel like I'm missing something about this logic though, can you explain what's happening here?

Reply:
If the last label is not in the `replicaLabels` list than no replica labels are set.
Replica labels are always at the end of the list so if the last one is not present in the `s.replicaLabels` map this guarantees that no replica labels need to be stripped.

If the last labels in the full list is a replica label than strip all replica labels by removing `len(s.replicaLabels)` labels from the full labels list.

Reply:
Worth to comment this indeed.